ghaggin/invariant-ekf
# Test Filter

Generates fake data and tests the Matlab implementation.

Primary work in:
  - ```gen_fake_data.m```
  - ```fake_data_test.m```

Key Additions:
1. Make fake data
    * Generate ground truth rotation and position from random polynomials
    * Take the derivatives to get the world frame acceleration and rates
    * Add gravity bias and rotate into the body frame
    * Return Imu "measurements" and ground truth
2. Test the filter using the fake data
    * Run prediction with fake imu "measurement"
    * Run correction with the ground truth data at a (possibly) reduced frequency
    * Plot filter output vs the ground truth to visually observe effectiveness

I also merged Sean and Juhan's separate fixes and moved everything to the ```scripts``` folder.

I tried to fix the ```runLeft.m``` and ```TestLIEKF.m``` functions but may have missed a thing or two.
